enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,439 @@
+as well as a new field `FilesToDelete` which is just a list of file paths:
+```yaml
+{
+  Services: []Service{...},
+  Files: []FileInfo{
+    "path": "filepath",
+    "checksum": "checksum of file, generated at compile time",
+    "content": "base64 encoded file data, similarly to how MCO stores files in MachineConfigs",
+  },
+  FilesToDelete: []string{...} ,
+  EnvironmentVars: map[string]string{...},
+  WatchedEnvironmentVars: []string{...},
+}
+```
+
+We can utilize [registry runtime-utils](https://github.com/openshift/runtime-utils/blob/master/pkg/registries/registries.go)
+to easily process and merge the contents of the IDMS/ITMS resources, [as MCO does](https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/container-runtime-config/helpers.go#L407).
+
+For registry config, I don't forsee any of the files having sensitive information. If new information arises suggesting
+otherwise, we can leverage our `crypto.go` library to safely encrypt the file content before placing it in the
+windows-services ConfigMap.
+
+The directory to save the generated registry config files is defined by WMCO -- it should be the same as what it set for
+the containerd_conf.toml's `plugins."io.containerd.grpc.v1.cri".registry.config_path`. 
+
+#### File Creation
+
+WICD, which already watches the windows-services ConfigMap, will react to events on it by copying over the newly
+generated registry config files to each Windows instance, if a file has changed.
+
+The containerd daemon [picks up the registry configuration without requiring restart](https://github.com/containerd/containerd/blob/main/docs/hosts.md#registry-configuration---introduction) so no further action required.
+
+#### File Cleanup
+
+WICD will look at the `FilesToDelete` field of the windows-services ConfigMap and ensure files at those locations are deleted.
+
+Mirror registry directory and files will be deleted on node cleanup as part of WMCO's `RemoveFilesAndNetworks()` function.
+
+#### Working example
+
+Example: A cluster contains the following resources
+```yaml
+apiVersion: config.openshift.io/v1 
+kind: ImageDigestMirrorSet 
+metadata:
+  name: ubi9repo
+spec:
+  imageDigestMirrors: 
+  - mirrors:
+    - example.io/example/ubi-minimal
+    - example.com/example2/ubi-minimal
+    source: registry.access.redhat.com/ubi9/ubi-minimal 
+    mirrorSourcePolicy: AllowContactingSource 
+  - mirrors:
+    - mirror.example.com
+    source: registry.redhat.io
+    mirrorSourcePolicy: NeverContactSource
+```
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageTagMirrorSet
+metadata:
+  name: tag-mirror
+spec:
+  imageTagMirrors:
+  - mirrors:
+    - docker.io
+    source: docker-mirror.internal
+    mirrorSourcePolicy: AllowContactingSource
+```
+
+would result in WMCO generating content for 3 files to transfer onto each Windows instance. The new windows-services
+ConfigMap that WMCO creates will have:
+
+```yaml
+{
+  Services: []Service{...},
+  Files: []FileInfo{
```

Comment:
Even after looking at the example I am not clear why this toml file is being treated differently when compared to the other config files for the node.